core/contract/src/main/java/org/eclipse/dataspaceconnector/contract/ContractNegotiationCommandExtension.java

```diff
@@ -30,23 +29,18 @@
 @CoreExtension
 @Provides({CommandHandlerRegistry.class})
 public class ContractNegotiationCommandExtension implements ServiceExtension {
-    
+
     @Inject
     private ContractNegotiationStore store;
-    
+
     @Override
     public void initialize(ServiceExtensionContext context) {
-        CommandHandlerRegistry registry = context.getService(CommandHandlerRegistry.class, true);
-        if (registry == null) {
-            registry = new CommandHandlerRegistryImpl();
-            context.registerService(CommandHandlerRegistry.class, registry);
-        }
-        
+        CommandHandlerRegistry registry = context.getService(CommandHandlerRegistry.class);
```

Comment:
This could injected with `@Inject` annotation

Reply:
OK, let's do that in a subsequent PR because there is some other cleanup I want to do here that is related.